### ISSUE TITLE
admin_generator broken in 1.9

### ISSUE THREAD

**Issue by lllama** [django-extensions/django-extensions#784]
Running the admin_generator command gives me the following error:

`LookupError: No installed app with label '<module 'myapp.models' from '/path/to/myapp/models.pyc.`

It looks like the command is trying to use the repr of the models module, rather than the app name.

I'm using 1.6.1 BTW

**Comment by iho** [comment]
The same problem/

**Comment by night-crawler** [comment]
Python==3.4, Django==1.9.1.

```
# 328 lineno
installed_apps = dict((a.__name__.rsplit('.', 1)[0], a) for a in get_apps())
```
returns a dict like `{'service_rest_api': <module 'service_rest_api.models'}`
But `AdminApp` in `__iter__` uses `get_models_compat` that seems take an app name.

I don't know what's inside django-extensions box, so i fixed it inplace with:
`self.app.__name__.split('.')[0]` as argument for `get_models_compat`

**Comment by dkdndes** [comment]
+1

**Comment by marshalc** [comment]
I can confirm same issue. 

`manage.py admin_generator nhic.ingest` gets me the error (truncated) of: 
```
File "/Users/carl/.virtualenvs/oxford-processing/lib/python3.5/site-packages/django/apps/registry.py", line 155, in get_app_config
    raise LookupError(message)
LookupError: No installed app with label 'nhic.ingest'. Did you mean 'ingest'?
```
Whereas the command `manage.py admin_generator ingest` gets me the feedback of...
```
This command requires an existing app name as argument
Available apps:
    django.contrib.admin
    django.contrib.auth
    django.contrib.contenttypes
    django.contrib.sessions
    django.contrib.sites
    django_extensions
    nhic.ingest
```

... so you can see the circular logic here suggesting there's a bug. Unfortunately I don't have time presently to investigate and debug this (this is my first time trying to use this extension), but if you want a practice project/model, you can see the current state of what I'm using this with at [https://github.com/nhic-tra/oxford-processing/tree/1748b5406b9d17397b235ccd464016dc3327060c](https://github.com/nhic-tra/oxford-processing/tree/1748b5406b9d17397b235ccd464016dc3327060c)

This is under:

* Django (1.9.6)
* django-extensions (1.6.7)
* Python 3.5.1

**Comment by trbs** [comment]
@marshalc Does any of the two commits above fixes the problem for you ?

@all, anybody wants to put a PR together ?

**Comment by marctc** [comment]
Hi @trbs, my commit was already merged since version 1.6.2 and fixes the initial issue. The bug that @marshalc refers is that admin generator don't work if you are working with nested apps like `nhic.ingest`. I'll try to find the solution asap. 

Maybe we can open a new issue and close this.

**Comment by trbs** [comment]
@marctc ack yeah please open a new issue, thanks for the help !
